### PR TITLE
Student notebook, downsync dialog, and max attempts

### PR DESCRIPTION
**Pull request by frostyfan109**
- Adds websockets
- Adds popup dialog when student launches indicating which files were added during downsyncing
- Adds max attempts information about the assignment
- Adds popup dialog to confirm student wants to submit if they are on their final attempt

### CODE REVIEW

File: eduhelx_jupyterlab_student/handlers.py [hunk 1]
```diff
@@ -84,6 +87,45 @@ def config(self) -> ExtensionConfig:
...
+        return JupyterHandler.prepare(self, *args, **kwargs)
+
+    @ws_authenticated
+    async def open(self):
+        if self not in self.clients: self.clients.append(self)
+        while len(self.queued_messages) > 0:
+            message = self.queued_messages.pop()
+            self.emit(*message[0], **message[1])
+
+    def on_message(self, message):
```
Comment (Hoid) [inline eduhelx_jupyterlab_student/handlers.py:117]: I don't think this is used anywhere else, and it's just printing

Comment (frostyfan109) [reply eduhelx_jupyterlab_student/handlers.py:117]: It's just debug at the moment, still a useful reference to leave in for now.

File: src/components/assignment-panel/assignment-info/assignment-info.tsx [hunk 2]
```diff
@@ -28,12 +29,12 @@ export const AssignmentInfo = ({  }: AssignmentInfoProps) => {
         let text = undefined
         let tooltip = undefined
         let filled = false
-        if (!assignment.isCreated) {
-            // Upcoming assignment (doesn't have either an open or close date yet)
+        if (!assignment.isCreated || !assignment.isAvailable) {
+            // Upcoming assignment (doesn't one either an open/close date OR open date not reached yet)
```
Comment (Hoid) [inline src/components/assignment-panel/assignment-info/assignment-info.tsx]: This comment doesn't make sense, might want to reword it 😆

Comment (frostyfan109) [reply src/components/assignment-panel/assignment-info/assignment-info.tsx]: fixed

**Review by Hoid — APPROVED** [review]

**Review by ptlharit2 — APPROVED** [review]

**Pull request merged**